replicaset.go

```diff
@@ -66,25 +61,44 @@ func Initiate(session *mgo.Session, address, name string, tags map[string]string
 	monotonicSession := session.Clone()
 	defer monotonicSession.Close()
 	monotonicSession.SetMode(mgo.Monotonic, true)
-	cfg := Config{
-		Name:    name,
-		Version: 1,
-		Members: []Member{{
-			Id:      1,
-			Address: fixIpv6Address(address),
-			Tags:    tags,
-		}},
+	// We don't know the mongod's ability to use a correct IPv6 addr format
+	// until the server is started, but we need to know before we can start
+	// it. Try the older, incorrect format if the correct format fails.
+	cfg := []Config{
+		Config{
+			Name:    name,
+			Version: 1,
+			Members: []Member{{
+				Id:      1,
+				Address: address,
+				Tags:    tags,
+			}},
+		},
+		Config{
+			Name:    name,
+			Version: 1,
+			Members: []Member{{
+				Id:      1,
+				Address: breakIpv6Address(address),
+				Tags:    tags,
+			}},
+		},
 	}
-	logger.Infof("Initiating replicaset with config %#v", cfg)
+
 	var err error
+	logger.Infof("Initiating replicaset with config %#v", cfg)
```

Comment:
should this actually be moved into the loop so we only report 1 config at a time?